### DISCUSSION TITLE
Should we add snap and/or flatpak support by default?

### DISCUSSION THREAD

**Discussion by boredland** [Manjaro-Sway/manjaro-sway#215] [Ideas]
Afaik a lot of the other manjaro distros have the companion packages for both installed, we don't without a conscious reason. Should we add them?

My personal opinion is, that we should enable both. If I am confronted with the decision to install from aur, repos, appimage, snap or flatpak, I usually check if the non-aur package is not outdated and go with snap / flatpak, whatever is newer, because of updates showing up for them in contrast to appImage iirc.

**Comment by boredland** [comment]
Regarding appImage we probably should install community/appimagelauncher as an extra by default.

**Comment by Mathew-D** [comment]
See I am the opposite., I always check the repos and if they don't have it then I use the aur package. The snap / flatpak packages always seem to add to much extra stuff. I could see maybe adding directions to the FAQ on how to install snap / flatpak if the user wants it but I am not sure we want to add more stuff by default.

**Reply by boredland** [reply]
My proposal would not be to enable that by default, see screenshots here: https://github.com/Manjaro-Sway/iso-profiles/pull/13

**Reply by Mathew-D** [reply]
Wouldn't that install snapd by default as well?

**Reply by boredland** [reply]
```
○ snapd.service - Snap Daemon
     Loaded: loaded (/usr/lib/systemd/system/snapd.service; disabled; vendor preset: disabled)
     Active: inactive (dead)
TriggeredBy: ○ snapd.socket
```

**Reply by boredland** [reply]
... and I marked them as extra. So what we could easily do is: offer minimal images again.

**Reply by boredland** [reply]
reintroduced minimal and unstable builds, reduced the amount of builds to weekly (sunday night) interval.

**Comment by nnuel** [comment]
I have enabled it both at times, but would always opt for the version in the AUR. Especially I find the flatpaks and snaps to clutter what I am looking for, so I mostly turn them off, but would use it if there's no other alternative.

Maybe they could be included, but disable them in the pamac-manager by default?

**Reply by boredland** [reply]
My proposal would not be to enable that by default, see screenshots here: Manjaro-Sway/iso-profiles#13

**Reply by nnuel** [reply]
This seems to be the same than as on Manjaro KDE, last I tried it, so it would be more uniform with users that want to try sway and expect some of the things to be there. But I'd also be fine with a FAQ entry saying what packages to install to enable Flatpak and/or Snap. I think the former might be more user-friendly, but maybe then steps could be written on how to remove it, if users wouldn't want it.

**Reply by boredland** [reply]
if you're using the pamac desktop app, filtering is pretty easy though.
![image](https://user-images.githubusercontent.com/4662748/148563106-a98970be-fbd0-4dec-a0b6-261827569131.png)

**Reply by nnuel** [reply]
Yes, it might be easy, but I like that the default ALL only shows stuff from the repository with a search. Something that you lose when you enable Snap (also Flatpak I assume). Like AUR is also excluded from searches in ALL, so if there was a setting like that or it would always standard search in Repositories, than it would be grand, but I don't believe there is. 

I do think Snaps and Flatpak integration are one of the stronger things about manjaro, like it's easy to have access to a great variety of software, so I'm all for including them by default in their disabled position!

**Reply by boredland** [reply]
Ya, I too think that is a good compromise.
